Update low prec codegen for div/mod

eellison opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #142350

Div/mod in fp16/bf16 requires a downcast to preserve its inputs' dtypes.


cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @yf225 @chenyang78 @kadeng @muchulee8 @ColinPeppler @amjames @desertfire @chauhang @aakhundov

eellison committed:
Update

[ghstack-poisoned]

eellison committed:
Update

[ghstack-poisoned]

blaine-rister commented on torch/_inductor/codegen/triton.py:
```diff
@@ -902,6 +900,25 @@ def constant(cls, value, dtype):
     def abs(x):
         return f"tl_math.abs({x})"
 
+    # TODO - register these ops as having divergent dtype
```
nit: Might be good to link to an issue to track the future direction of this.

blaine-rister reviewed (approved):
LGTM!

eellison committed:
Update

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison commented:
@pytorchbot merge

eellison commented:
@pytorchbot merge

eellison commented:
@pytorchbot merge -f "tests were fully passing and i accidentally closed"

eellison committed:
Update

[ghstack-poisoned]

eellison committed:
Update

[ghstack-poisoned]

eellison committed:
Update

[ghstack-poisoned]

eellison committed:
Update

[ghstack-poisoned]

eellison committed:
Update

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison committed:
Update

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison committed:
Update

[ghstack-poisoned]

eellison commented:
@pytorchbot merge

eellison commented:
@huydhn @blaine-rister was going to stack a pr updating the tolerances - if it's a single test failure would you mind checking before reverting ?

huydhn commented:
> @huydhn @blaine-rister was going to stack a pr updating the tolerances - if it's a single test failure would you mind checking before reverting ?

Oh ok, the diff has been in the train since last Friday, so I assumed that a revert/reland would be easier.  I'll keep this in mind.

huydhn commented:
@pytorchbot merge -f 'Reland this as there is a forward fix coming'

huydhn commented:
@blaine-rister  I have re-imported the PR into D67301180, please feel free to rebase your forward on top of that one

jeanschmidt commented:
@pytorchbot revert -m "breaking internal signals" -c ghfirst

eellison commented:
@pytorchbot rebase

eellison committed:
Update on "Update low prec codegen for div/mod"


Div/mod in fp16/bf16 requires a downcast to preserve its inputs' dtypes.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

eellison committed:
Update on "Update low prec codegen for div/mod"


Div/mod in fp16/bf16 requires a downcast to preserve its inputs' dtypes.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

eellison committed:
Update on "Update low prec codegen for div/mod"


Div/mod in fp16/bf16 requires a downcast to preserve its inputs' dtypes.


cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx ipiszy yf225 chenyang78 kadeng muchulee8 ColinPeppler amjames desertfire chauhang aakhundov

[ghstack-poisoned]

eellison commented:
@pytorchbot merge
